src/AcceptanceTests/Infrastructure/Cleaner.cs

```diff
@@ -0,0 +1,56 @@
+﻿namespace NServiceBus.AzureServiceBus.AcceptanceTests.Infrastructure
+{
+    using System;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Logging;
+    using Microsoft.ServiceBus;
+    using Microsoft.ServiceBus.Messaging;
+    using NServiceBus.AcceptanceTests;
+    using NUnit.Framework;
+
+    public class Cleaner : NServiceBusAcceptanceTest
+    {
+        [Test, Explicit("Intended to be executed explicitly to delete all queues and topics.")]
+        [Category("Cleanup")]
+        public async Task DeleteEntities()
+        {
+            var namespaceManager = NamespaceManager.CreateFromConnectionString(TestUtility.GetDefaultConnectionString());
+
+            var queryQueues = namespaceManager.GetQueuesAsync();
+            var queryTopics = namespaceManager.GetTopicsAsync();
+            await Task.WhenAll(queryQueues, queryTopics)
+                .ConfigureAwait(false);
+
+            var deleteQueues = queryQueues.Result.Select(queueDescription => namespaceManager.DeleteQueueAsync(queueDescription.Path));
+            var deleteTopics = queryTopics.Result.Select(topicDescription => namespaceManager.DeleteTopicAsync(topicDescription.Path));
+
+            await TryWithRetries(Task.WhenAll(deleteQueues.Concat(deleteTopics)), 5)
+                .ConfigureAwait(false);
+        }
+
+        static async Task TryWithRetries(Task task, int maxRetryAttempts, int usedRetryAttempts = 0)
```

Comment:
We have this helper method https://github.com/Particular/NServiceBus.AzureServiceBus/blob/db4a1d9ca44f59e6d95f47368e3716826abf4d14/src/Transport/Connectivity/TaskWithRetryExtensions.cs#L10 which might come in handy?

Reply:
It's on `IMessageSender` which I don't need.

Reply:
I should have been more clear. I meant have you looked at how this method is implemented and reused what you needed?

Reply:
I've used a similar pattern indeed.